File: src/test/java/kitchenpos/tobe/eatinorders/application/ordertable/FakeEatInOrderRepository.java
package kitchenpos.tobe.eatinorders.application.ordertable;

import static kitchenpos.tobe.eatinorders.application.Fixtures.NOT_COMPLETED_ORDER_TABLE_ID;

import java.util.List;
import java.util.Optional;
import java.util.UUID;
import kitchenpos.eatinorders.tobe.domain.model.EatInOrder;
import kitchenpos.eatinorders.tobe.domain.model.EatInOrderStatus;
import kitchenpos.eatinorders.tobe.domain.repository.EatInOrderRepository;

public class FakeEatInOrderRepository implements EatInOrderRepository {

    @Override
    public EatInOrder save(EatInOrder eatInOrder) {
        return null;
    }

    @Override
    public Optional<EatInOrder> findById(UUID id) {
        return Optional.empty();
    }

    @Override
    public List<EatInOrder> findAll() {
        return null;
    }

    @Override
    public boolean existsByOrderTableIdAndStatusNot(UUID orderTableId, EatInOrderStatus status) {
        if (orderTableId.equals(NOT_COMPLETED_ORDER_TABLE_ID)) {
            return true;
        }
        return false;
    }
}

File: src/test/java/kitchenpos/eatinorders/tobe/FakeEatInOrderRepository.kt
package kitchenpos.eatinorders.tobe

import kitchenpos.eatinorders.tobe.domain.EatInOrder
import kitchenpos.eatinorders.tobe.domain.EatInOrderRepository
import kitchenpos.eatinorders.tobe.domain.EatInOrderStatus
import java.util.*

object FakeEatInOrderRepository : EatInOrderRepository {
    private val eatInOrderMap: MutableMap<UUID, EatInOrder> = mutableMapOf()

    override fun save(eatInOrder: EatInOrder): EatInOrder {
        eatInOrderMap[eatInOrder.id] = eatInOrder

        return eatInOrder
    }

    override fun existsByOrderTableIdAndStatusNot(orderTableId: UUID, status: EatInOrderStatus): Boolean {
        return eatInOrderMap.filterValues { it.status != status }.isNotEmpty()
    }

    override fun findById(eatInOrderId: UUID): Optional<EatInOrder> {
        return Optional.ofNullable(eatInOrderMap[eatInOrderId])
    }
}
